pallets/staking/src/lib.rs

```diff
@@ -0,0 +1,261 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_election_provider_support::{
+	ElectionDataProvider, ElectionProvider, ElectionProviderBase, SortedListProvider, Supports,
+};
+use frame_support::defensive;
+use frame_support::{traits::Defensive, RuntimeDebug};
+#[cfg(feature = "std")]
+pub use pallet_staking_substrate::GenesisConfig;
+#[cfg(feature = "std")]
+pub use pallet_staking_substrate::TestBenchmarkingConfig;
+use pallet_staking_substrate::ValidatorPrefs;
+pub use pallet_staking_substrate::{
+	ActiveEra, ActiveEraInfo, Config, ErasStartSessionIndex, Error, Event, ForceEra, Forcing,
+	Pallet, RewardDestination, UseValidatorsMap,
+};
+pub use pallet_staking_substrate::{
+	__InherentHiddenInstance, __substrate_call_check, __substrate_event_check,
+	__substrate_genesis_config_check, tt_default_parts, tt_error_token,
+};
+use parity_scale_codec::{Decode, Encode, EncodeLike};
+use scale_info::TypeInfo;
+use serde::{Deserialize, Serialize};
+use sp_runtime::traits::{OpaqueKeys, Zero};
+use sp_runtime::AccountId32;
+pub use sp_staking::{EraIndex, StakingInterface};
+use sp_std::{boxed::Box, fmt::Debug, marker::PhantomData, vec, vec::Vec};
+
+pub(crate) const LOG_TARGET: &str = "runtime::staking";
+
+macro_rules! logger {
+	($level:tt, $patter:expr $(, $values:expr)* $(,)?) => {
+		log::$level!(
+			target: crate::LOG_TARGET,
+			concat!("[{:?}] 💸 ", $patter), <frame_system::Pallet<T>>::block_number() $(, $values)*
+		)
+	};
+}
+pub(crate) use logger;
+
+pub mod weights {
+	pub use pallet_staking_substrate::weights::*;
+}
+
+pub struct EmptyList<T>(PhantomData<T>);
```

Comment:
I see lots of uncovered lines of code but I'm not sure what they are meant to be. 

What's the idea behind the `EmptyList` and `TrivialTargetList` structs? 

I'm also not sure how the currently proposed implementation changes the scope of required testing but I'll leave this for later.

Reply:
There are some core associated types around maintaining and managing consensus data structure that we don't immediately need. These are not optional so we can't use `()` instead. We could probably merge them both but I would rather have them there being annoying than introducing new propositions into the system.